sources_test.go

```diff
@@ -26,6 +27,42 @@ func TestFile(t *testing.T) {
 	}
 }
 
+func TestListFilesMultipleFiles(t *testing.T) {
+	t.Parallel()
+	dir := "testdata/multiple_files"
+	want := fmt.Sprintf("%s/1.txt\n%s/2.txt\n%s/3.tar.zip", dir, dir, dir)
+	p := ListFiles(dir)
+	got, err := p.String()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got != want {
+		t.Errorf("In %q: want %q, got %q", dir, want, got)
+	}
+}
+
+func TestListFilesNonexistent(t *testing.T) {
+	t.Parallel()
+	p := ListFiles("nonexistentpath")
+	if p.Error() == nil {
+		t.Errorf("want error status on listing non-existent path, but got nil")
+	}
+}
+
+func TestListFilesGlob(t *testing.T) {
+	t.Parallel()
+	dir := "testdata/multiple_files"
+	want := fmt.Sprintf("%s/1.txt\n%s/2.txt", dir, dir)
+	p := ListFiles("testdata/multi?le_files/*.txt")
+	got, err := p.String()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if want != got {
+		t.Errorf("In %q: want %q, got %q", dir, want, got)
```

Comment:
Both `want` and `got` already contain the `dir` component, so we don't need to repeat it here.

Reply:
Fixed this one